### DISCUSSION TITLE
45. 네트워크 계층과 데이터 링크 계층의 차이점은 무엇인가요?

### DISCUSSION THREAD

**Discussion by linirini** [jewelry-box/cs-interview#46] [General]
.

**Comment by eun-byeol** [comment]
가장 큰 차이점은 `전송 범위`입니다. `네트워크 계층`은 `서로 다른 네트워크 간` 데이터를 전송할 수 있는 반면, `데이터 링크 계층`은 `같은 네트워크 내` 데이터를 전송합니다.
또한, 데이터 전송을 위해 네트워크 계층은 `논리 주소`인 `IP`를 사용하고 데이터 링크 계층은 `물리 주소`인 `MAC 주소`를 사용합니다.

---
> 네트워크 vs 데이터 링크

특징 | 네트워크 계층 | 데이터 링크 계층
-- | -- | --
주소 | 논리 주소(IP 주소) | 물리 주소(MAC 주소)
전송 단위 | 패킷(Packet) | 프레임(Frame)
전송 범위 | 다른 네트워크 간 데이터 전송 | 같은 네트워크 내 데이터 전송
주요 프로토콜 | IP, ARP*, ICMP* | Ethernet*, Wi-Fi*, PPP*
장비 | 라우터 | 스위치, 브리지
오류 처리 | 패킷 손실은 상위 계층(TCP)에서 처리 | 프레임 손상 여부를 확인하고 재전송 요청 가능

- ICMP(Internet Control Message Protocol): 네트워크 문제 진단 및 오류 메시지 전달
- ARP(Address Resolution Protocol): IP 주소를 MAC 주소로 변환
- Ethernet: 유선 LAN에서 널리 사용
- Wi-Fi: 무선 네트워크에서 사용
- PPP(Point-to-Point Protocol): 두 노드 간 데이

**Reply by cutehumanS2** [reply]
PPP는 데이터링크 계층에서 두 노드 간(Point-to-Point) 신뢰성있는 데이터를 전송하기 위해 사용되는 통신 프로토콜입니다.

-> 설명이 잘렸길래 ㅋ.ㅋ
